internal/orchestrator/committer.go

```diff
@@ -70,17 +70,19 @@ func (c *Committer) Start() {
 }
 
 func (c *Committer) getBlockNumbersToCommit() ([]*big.Int, error) {
-	maxBlockNumber, err := c.storage.MainStorage.GetMaxBlockNumber()
+	latestCommittedBlockNumber, err := c.storage.MainStorage.GetMaxBlockNumber()
+	log.Info().Msgf("Committer found this max block number in main storage: %s", latestCommittedBlockNumber.String())
 	if err != nil {
 		return nil, err
 	}
 
-	if maxBlockNumber.Cmp(big.NewInt(0)) == 0 {
-		maxBlockNumber = new(big.Int).Sub(c.pollFromBlock, big.NewInt(1))
+	if latestCommittedBlockNumber.Cmp(big.NewInt(0)) == 0 {
+		// If no blocks have been committed yet, start from the fromBlock specified in the config (same start for the poller)
+		latestCommittedBlockNumber = c.pollFromBlock
```

Comment:
You need to subtract 1 here, since the next line will add 1 to the `latestCommitedBlockNumber` as the start block, since that number is already commited and you want the next.